models/contacts.js

```diff
@@ -1,19 +1,102 @@
-// const fs = require('fs/promises')
+import path from "path";
+import fs from "fs/promises";
+import { nanoid } from "nanoid";
 
-const listContacts = async () => {}
+const contactsPath = path.join(process.cwd(), "/models/contacts.json");
 
-const getContactById = async (contactId) => {}
+export const listContacts = async () => {
+  try {
+    const contacts = await fs.readFile(contactsPath);
+    const contactsParsed = JSON.parse(contacts);
+    return contactsParsed;
+  } catch (error) {
+    return JSON.parse(error.message);
+  }
+};
 
-const removeContact = async (contactId) => {}
+export const getById = async (contactId) => {
+  try {
+    const contacts = await fs.readFile(contactsPath);
 
-const addContact = async (body) => {}
+    const contactsParsed = JSON.parse(contacts);
+    const [contact] = contactsParsed.filter(
```

Comment:
Możesz użyć find metody, jeśli nie znajdzie zwróci undefined

Reply:
Proszę o doprecyzowanie komentarza, nie bardzo rozumiem co mam zmienić.  W funkcji addContact korzystam z metody find.

Reply:
Nie trzeba zmianiać, tylko tutaj można użyć metody find. Bo używasz filter.
```suggestion
    const contact = contactsParsed.find(c => c.id === contactId);
```
Find zwraca undefined i nie musisz potem robić `if index >  -1`

Reply:
Rozumiem, zmiany wprowadzone, dziękuję za cenną wskazówkę :)